packages/wrangler/src/deployment-bundle/node-compat.ts

```diff
@@ -0,0 +1,103 @@
+import { UserError } from "../errors";
+import { logger } from "../logger";
+
+export type NodeJSCompatMode = "legacy" | "v1" | "v2" | null;
+
+/**
+ * Validate and compute the Node.js compatibility mode we are running.
+ *
+ * Returns one of:
+ *  - "legacy": build-time polyfills, from `node_compat` flag
+ *  - "v1": nodejs_compat compatibility flag
+ *  - "v2": experimental nodejs_compat_v2 flag
+ *  - null: no Node.js compatibility
+ *
+ * Currently we require that the v2 mode is configured via `experimental:nodejs_compat_v2` compat flag,
+ * where the `experimental:` prefix is stripped before being passed to the runtime since that does not
+ * understand this prefix.
+ *
+ * We assert that only one of these modes can be specified at a time.
+ * We assert that you must prefix v2 mode with `experimental`.
+ * We warn if using legacy or v2 mode.
+ */
+export function validateNodeCompat({
+	legacyNodeCompat,
+	compatibilityFlags,
+	noBundle,
+}: {
+	legacyNodeCompat: boolean;
+	/* mutate */ compatibilityFlags: string[];
+	noBundle: boolean;
+}): NodeJSCompatMode {
+	if (legacyNodeCompat) {
+		logger.warn(
+			"Enabling Wrangler compile-time Node.js compatibility polyfill mode for builtins and globals. This is experimental and has serious tradeoffs."
+		);
+	}
+
+	const nodejsCompatV2 = compatibilityFlags.includes(
+		"experimental:nodejs_compat_v2"
+	);
+	const nodejsCompatV2NotExperimental =
+		compatibilityFlags.includes("nodejs_compat_v2");
+
+	if (nodejsCompatV2) {
+		// strip the "experimental:" prefix because workerd doesn't understand it yet.
```

Comment:
Given the flag is no longer experimental in workerd, do we want to keep this prefix stripping?

Reply:
No we still want to require users of Wrangler to add this - it was orthogonal to workerd's experimental constraints - and is left here to avoid people trying to use this for production while we stabilize it.